geniza/corpus/models.py

```diff
@@ -764,7 +766,24 @@ def iiif_images(self, filter_side=False, with_placeholders=False):
                 for canvas_uri in ed.content_html.keys():
                     if canvas_uri not in iiif_images:
                         # use placeholder image for each canvas not in iiif_images
-                        iiif_images[canvas_uri] = Document.PLACEHOLDER_CANVAS
+                        iiif_images[canvas_uri] = deepcopy(Document.PLACEHOLDER_CANVAS)
+                        uri_match = re.search(
+                            r"textblock\/(\d+)\/canvas\/(\d)\/", canvas_uri
+                        )
```

Comment:
regex seems like overkill, but I guess I can see the convenience. recommend using named matches so the group reference is more semantic when you pull textblock and canvas ids